lightning/src/chain/chainmonitor.rs

```diff
@@ -194,6 +194,35 @@ pub trait Persist<ChannelSigner: WriteableEcdsaChannelSigner> {
 	///
 	/// [`Writeable::write`]: crate::util::ser::Writeable::write
 	fn update_persisted_channel(&self, channel_funding_outpoint: OutPoint, update: Option<&ChannelMonitorUpdate>, data: &ChannelMonitor<ChannelSigner>, update_id: MonitorUpdateId) -> ChannelMonitorUpdateStatus;
+	/// Archive a channel's data to a backup location.
+	///
+	/// Invoked by [`ChainMonitor::archive_stale_channel_monitors`].
+	///
+	/// This function can be used as a first step in pruning a channel's monitor data.
+	/// The data should be moved to an archive location, and then removed
+	/// using [`Persist::remove_persisted_channel`].
+	///
+	/// This is usually called when a channel monitor is in a stale state, ie no
+	/// fund can be claimed with its data. This is a good time to
+	/// archive the channel data to a backup location, before removing
+	/// it from the primary storage.
+	///
+	/// Archiving the data is useful for hedging against data loss in
+	/// case of an unexpected failure/bug.
+	fn archive_persisted_channel(&self, channel_funding_outpoint: OutPoint) -> ChannelMonitorUpdateStatus;
```

Comment:
@TheBlueMatt I did split the archiving and removing process so together they can construct a `prune` process. that will allow us also to have the logic up in the `archive_stale_monitors` and not duplicate the code in a single function in the trait that would be responsible on doing more than one thing.